### ISSUE TITLE
External gene call - Eukaryota

### ISSUE THREAD

**Issue by Lcornet** [merenlab/anvio#1538]
## Short description of the problem

I would like to use "anvi-gen-contigs-database" for Fungi genomes for which I have Maker proteins predictions.
I have this error : 
The sequence corresponds to the gene callers id '1' does not seem to have proper number of nucleotides to be translated

## anvi'o version

6.1

## System info

Singularity

## Detailed description of the issue

From the GFF file of maker, in constructed a gene-call file for anvio: 
This how the GFF looks like : 
NW_003456411.1_RaGOO    maker   gene    33802   35186   .       +       .       ID=maker-NW_003456411.1_RaGOO-augustus-gene-0.30;Name=maker-NW_003456411.1_RaGOO-augustus-gene-0.30
NW_003456411.1_RaGOO    maker   mRNA    33802   35186   .       +       .       ID=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1;Parent=maker-NW_003456411.1_RaGOO-augustus-gene-0.30;Name=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1;_AED=0.01;_eAED=0.01;_QI=0|0|0|1|1|1|2|0|440
NW_003456411.1_RaGOO    maker   exon    33802   33968   .       +       .       ID=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1:exon:0;Parent=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1
NW_003456411.1_RaGOO    maker   exon    34031   35186   .       +       .       ID=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1:exon:1;Parent=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1
NW_003456411.1_RaGOO    maker   CDS     33802   33968   .       +       0       ID=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1:cds;Parent=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1
NW_003456411.1_RaGOO    maker   CDS     34031   35186   .       +       1       ID=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1:cds;Parent=maker-NW_003456411.1_RaGOO-augustus-gene-0.30-mRNA-1

I took the mRNA sequence  to produce:

gene_callers_id contig  start   stop    direction       partial source  version
1       NW_003456411.1_RaGOO    33801   35186   f       0       maker    V1.0

But I have this error:

Config Error: The sequence corresponds to the gene callers id '1' does not seem to have proper
              number of nucleotides to be translated :/

I wonder how to create an external gene call for eukaryote as the aa_sequence field is no longer usable.
Can you help me ?

Thanks,
Luc

**Comment by Charlotte-Nef** [comment]
Hi Luc,

I had the same problem when working with other eukaryotes (microalgae), because anvi'o was first developed for prokaryotic models. I tried to import exons that were not all divisible by three yet anvi'o requires them to be divisible by three to predict the codons when importing the gene calls.

We finally found a provisional solution by removing one or two nucleotides at the end of the sequences to make the exons divisible by 3 and import them into anvi'o. This is not an optimal solution as it will prevent us from working on SAAVs (because the frame is shifted) but it allowed us to compute the variability tables and export SNVs for our samples. So I suggest that you try this method and see whether your gene calls can be imported. For your information I'm working on the anvi'o master branch.

Hope this helps!

Best,

Charlotte
